pylint/checkers/base.py

```diff
@@ -1524,14 +1524,11 @@ def _check_reversed(self, node):
                 return
 
             if isinstance(argument, astroid.Instance):
-                if argument._proxied.name == "dict" and utils.is_builtin_object(
-                    argument._proxied
-                ):
-                    self.add_message("bad-reversed-sequence", node=node)
-                    return
                 if any(
                     ancestor.name == "dict" and utils.is_builtin_object(ancestor)
-                    for ancestor in argument._proxied.ancestors()
+                    for ancestor in itertools.chain(
+                        (argument._proxied,), argument._proxied.ancestors()
+                    )
                 ):
```

Comment:
Instead of a check for the system version, I removed the hardcoded `add_message` on dicts. We should be able to find the `__reversed__` dunder on py3.8 but not >py3.7, which i think should address this bug in a cleaner way IMO. I *think* the performance should be the same, but LMK if this seems too slow, or if there's a reason why this might not work!